File: WebApp/indoor-localization/src/services/FacilityService.ts
import { Facility } from "../entities/Facility";
import { EntityService } from "./Service";

export class FacilityService extends EntityService {

  static async Get(id: number): Promise<Facility | null> {
    return this.facilityRepo.Get(id);
  }

  static async GetAll(): Promise<Facility[]> {
    return this.facilityRepo.GetAll();
  }

  static async Add(facility: Facility): Promise<boolean> {
    return this.facilityRepo.Add(facility);
  }

  static async Delete(id: number): Promise<boolean> {
    return this.facilityRepo.Delete(id);
  }

  static async Update(facility: Facility): Promise<boolean> {
    return this.facilityRepo.Update(facility);
  }
}

File: Android/indoor_localization/lib/domain/services/facility_service.dart
import 'package:indoor_localization/domain/entities/asset.dart';
import 'package:indoor_localization/domain/entities/facility.dart';
import 'package:indoor_localization/domain/entities/zone.dart';
import 'package:indoor_localization/domain/services/entity_service.dart';

class FacilityService extends EntityService {

  static Future<Facility?> get(int id) async {
    return await EntityService.facilityRepo.get(id);
  }

  static Future<List<Facility>> getAll() {
    return EntityService.facilityRepo.getAll();
  }

  static bool add(Facility facility) {
    return EntityService.facilityRepo.add(facility);
  }

  static bool delete(int id) {
    return EntityService.facilityRepo.delete(id);
  }

  static bool update(Facility facility) {
    return EntityService.facilityRepo.update(facility);
  }
}